internal/flow/describe_release.go

```diff
@@ -0,0 +1,119 @@
+package flow
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/lunarway/release-manager/internal/artifact"
+	"github.com/lunarway/release-manager/internal/commitinfo"
+	"github.com/lunarway/release-manager/internal/git"
+	"github.com/lunarway/release-manager/internal/intent"
+	"github.com/lunarway/release-manager/internal/log"
+	"github.com/pkg/errors"
+)
+
+type DescribeReleaseResponse struct {
+	Releases []DescribeReleaseResponseRelease
+}
+
+type DescribeReleaseResponseRelease struct {
+	DefaultNamespaces bool
+	ReleaseIndex      int
+	Artifact          artifact.Spec
+	ReleasedAt        time.Time
+	ReleasedByEmail   string
+	ReleasedByName    string
+	Intent            intent.Intent
+}
+
+// DescribeRelease returns information about a specific release in an environment.
+func (s *Service) DescribeRelease(ctx context.Context, environment, service string, count int) (DescribeReleaseResponse, error) {
+	span, ctx := s.Tracer.FromCtx(ctx, "flow.DescribeRelease")
+	span.SetBaggageItem("count", fmt.Sprintf("%v", count))
+
+	defer span.Finish()
+	sourceConfigRepoPath, close, err := git.TempDirAsync(ctx, s.Tracer, "k8s-config-describe-release")
+	if err != nil {
+		return DescribeReleaseResponse{}, err
+	}
+	defer close(ctx)
+
+	log.WithContext(ctx).Debugf("Cloning source config repo %s into %s", s.Git.ConfigRepoURL, sourceConfigRepoPath)
+	sourceRepo, err := s.Git.Clone(ctx, sourceConfigRepoPath)
+	if err != nil {
+		return DescribeReleaseResponse{}, errors.WithMessagef(err, "clone into '%s'", sourceConfigRepoPath)
+	}
+
+	var releases []DescribeReleaseResponseRelease
+
+	var currentOffset uint = 0
+	for count-int(currentOffset) > 0 {
```

Comment:
Could this be a simple `for` loop instead? I can't see why we need to track the offset like this?

```suggestion
	for currentOffset := 0; currentOffset++; currentOffset < count {
```

Reply:
Could, and did consider that, but found the current easier to understand. But I'll change it